src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt

```diff
@@ -233,4 +213,39 @@ interface VisitRepository : JpaRepository<OldVisit, Long>, JpaSpecificationExecu
     newSessionTemplateReference: String,
     fromDate: LocalDate,
   ): Int
+
+  @Query(
+    "SELECT v FROM Visit v JOIN v.visitors vis WHERE " +
+      "(:#{#filter.prisonerId} is null OR v.prisonerId = :#{#filter.prisonerId})  AND " +
+      "(:#{#filter.prisonCode} is null OR v.prison.code = :#{#filter.prisonCode}) AND " +
+      "(:#{#filter.visitStatusList} is null OR :#{#filter.visitStatusList.size()==0} OR v.visitStatus in :#{#filter.visitStatusList}) AND " +
+      "(:#{#filter.visitorId} is null OR vis.nomisPersonId = :#{#filter.visitorId}) AND " +
+      "(:#{#filter.startDateTime} is null OR (v.sessionSlot.slotDate >= :#{#filter.startDateTime.toLocalDate()} AND v.sessionSlot.slotTime >= :#{#filter.startDateTime.toLocalTime()})) AND " +
+      "(:#{#filter.endDateTime} is null OR (v.sessionSlot.slotDate <= :#{#filter.endDateTime.toLocalDate()} AND v.sessionSlot.slotEndTime < :#{#filter.endDateTime.toLocalTime()})) " +
+      " ORDER BY v.sessionSlot.slotDate,v.sessionSlot.slotTime",
+  )
+  fun findVisitsOrderByDateAndTime(filter: VisitFilter, pageable: Pageable): Page<Visit>
```

Comment:
we could revisit the code and see if we need to keep the queries for start date and time and end date and time or we can replace it with a date

Reply:
I think get it all running revisit this when tests running.